lms/djangoapps/instructor_task/tasks_helper/misc.py

```diff
@@ -466,3 +467,57 @@ def upload_ora2_submission_files(
     TASK_LOG.info('%s, Task type: %s, Upload complete.', task_info_string, action_name)
 
     return UPDATE_STATUS_SUCCEEDED
+
+
+def generate_anonymous_ids(_xmodule_instance_args, _entry_id, course_id, task_input, action_name):  # lint-amnesty, pylint: disable=too-many-statements
+    """
+    Generate a 2-column CSV output of user-id, anonymized-user-id
+    """
+    def _log_and_update_progress(step):
+        """
+        Updates progress task and logs
+
+        Arguments:
+            step: current step task is on
+        """
+        TASK_LOG.info(
+            '%s, Task type: %s, Current step: %s for all learners',
+            task_info_string,
+            action_name,
+            step,
+        )
+        task_progress.update_task_state(extra_meta=step)
+    TASK_LOG.info('Testing log: Starting task execution for generation of Anonymize IDs report.')
+
+    task_info_string_format = 'Task: {task_id}, InstructorTask ID: {entry_id}, Course: {course_id}, Input: {task_input}'
+    task_info_string = task_info_string_format.format(
+        task_id=_xmodule_instance_args.get('task_id') if _xmodule_instance_args is not None else None,
+        entry_id=_entry_id,
+        course_id=course_id,
+        task_input=task_input
+    )
+    TASK_LOG.info('%s, Task type: %s, Starting task execution', task_info_string, action_name)
+
+    start_time = time()
+    start_date = datetime.now(UTC)
+
+    students = User.objects.filter(
+        courseenrollment__course_id=course_id,
+    ).order_by('id')
+
+    task_progress = TaskProgress(action_name, students.count, start_time)
+    current_step = {'step': "Started Computing rows"}
+    _log_and_update_progress(current_step)
```

Comment:
Once you have updated the `current_step` variable, `_log_and_update_progress` method will use it, you do not need to pass. it.

Reply:
or even better, pass it to `_log_and_update_progress` directly. 
`_log_and_update_progress("Compiling learner rows")`

--

and then
 `task_progress.update_task_state(extra_meta={'step': step})`